src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2235 @@
+    dst_f8->f1[3] = fmaf(src_f1, filter[2], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[1], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[0], dst_f8->f1[5]);
+    src_f1 = rpp_hip_unpack2(src_ui4.y);
+    dst_f8->f1[0] = fmaf(src_f1, filter[6], dst_f8->f1[0]);
+    dst_f8->f1[1] = fmaf(src_f1, filter[5], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter[4], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[3], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[2], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[1], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[0], dst_f8->f1[6]);
+    src_f1 = rpp_hip_unpack3(src_ui4.y);
+    dst_f8->f1[1] = fmaf(src_f1, filter[6], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter[5], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[4], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[3], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[2], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[1], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[0], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack0(src_ui4.z);
+    dst_f8->f1[2] = fmaf(src_f1, filter[6], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[5], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[4], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[3], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[2], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[1], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack1(src_ui4.z);
+    dst_f8->f1[3] = fmaf(src_f1, filter[6], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[5], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[4], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[3], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[2], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack2(src_ui4.z);
+    dst_f8->f1[4] = fmaf(src_f1, filter[6], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[5], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[4], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[3], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack3(src_ui4.z);
+    dst_f8->f1[5] = fmaf(src_f1, filter[6], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[5], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[4], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack0(src_ui4.w);
+    dst_f8->f1[6] = fmaf(src_f1, filter[6], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[5], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack1(src_ui4.w);
+    dst_f8->f1[7] = fmaf(src_f1, filter[6], dst_f8->f1[7]);
+}
+
+__device__ void gaussian_filter_9x9_row_hip_compute(uchar *srcPtr, d_float8 *dst_f8, float *filter)
+{
+    float src_f1;
+    uint4 src_ui4 = *(reinterpret_cast<uint4 *>(srcPtr));
+    src_f1 = rpp_hip_unpack0(src_ui4.x);
+    dst_f8->f1[0] = fmaf(src_f1, filter[0], dst_f8->f1[0]);
```

Comment:
why can't we do vector math for this fmaf instructions?

Reply:
@r-abishek @rrawther Yes, that's correct. The fmaf works with float, not float4.